File: Benchmark_1/scala/src/BenchMark.scala
import java.io.{BufferedReader, FileReader, IOException}

object BenchMark {
  private def sameContent(words1: String, words2: String) = {
    val line1 : Array[Char] = words1.toCharArray()
    val line2 : Array[Char] = words2.toCharArray()
    java.util.Arrays.parallelSort(line1)
    java.util.Arrays.parallelSort(line2)
    java.util.Arrays.equals(line1, line2)
  }

  @throws[IOException]
  def main(args: Array[String]): Unit = {
    val bufferedReader = new BufferedReader(new FileReader("../input_2.txt"))
    val words1 : String = bufferedReader.readLine
    val words2 : String = bufferedReader.readLine
    bufferedReader.close()
    //System.out.println("words1 = " + words1.mkString(" ") + ", size = " + words1.length)
    //System.out.println("words2 = " + words2.mkString(" ") + ", size = " + words2.length)
    val result1 = sameContent(words1, words2)
    System.out.println("SCALA")
    System.out.println("result1 = " + result1)
  }
}

File: Benchmark_1/java/src/BenchMark.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.Arrays;

public class BenchMark {

    private static boolean sameContent(final String words1, final String words2) {
        final char[] line1 = words1.toCharArray();
        final char[] line2 = words2.toCharArray();
        Arrays.parallelSort(line1);
        Arrays.parallelSort(line2);
        return Arrays.equals(line1, line2);
    }

    public static void main(final String[] args) throws IOException {
        final BufferedReader bufferedReader = new BufferedReader(new FileReader("../input_2.txt"));

        final String words1 = bufferedReader.readLine();
        final String words2 = bufferedReader.readLine();
        bufferedReader.close();

        //System.out.println("words1 = " + Arrays.toString(words1) + ", size = " + words1.length);
        //System.out.println("words2 = " + Arrays.toString(words2) + ", size = " + words2.length);

        final boolean result1 = sameContent(words1, words2);

        System.out.println("JAVA");
        System.out.println("result1 = " + result1);
    }
}
